distributed/tests/test_worker.py

```diff
@@ -1755,3 +1756,30 @@ async def test_taskstate_metadata(cleanup):
 
                 # Check that Scheduler TaskState.metadata was also updated
                 assert s.tasks[f.key].metadata == ts.metadata
+
+
+@pytest.mark.asyncio
+async def test_executor_offload(cleanup, monkeypatch):
+    class SameThreadClass:
+        def __getstate__(self):
+            return ()
+
+        def __setstate__(self, state):
+            self._thread_ident = threading.get_ident()
+            return self
+
+    monkeypatch.setattr("distributed.worker.OFFLOAD_THRESHOLD", 1)
```

Comment:
I'm not used to `monkeypatch`.  Does this clear out at the end of this test automatically, or do we need to do a try-finally here?

Reply:
It clears at the end of this test